src/gwt/src/org/rstudio/core/client/command/ShortcutManager.java

```diff
@@ -281,11 +275,8 @@ public void onPreviewNativeEvent(NativePreviewEvent event)
          if (handleKeyDown(event.getNativeEvent()))
          {
             event.cancel();
-            resetKeyBuffer();
-         }
-         else
-         {
-            updateKeyBuffer(event.getNativeEvent());
+            keyBuffer_.clear();
+            events_.fireEvent(new RStudioCommandExecutedEvent());
```

Comment:
This does need to be executed from the preview handler (to ensure Ace doesn't see commands handled by RStudio)